### PR TITLE
Returning basictests to its original function: basic tests (only)

### PR DESCRIPTION
**Pull request by lemire**
The initial motivation behind basictests was for a quick set of sanity tests to check whether your code made sense. It was not meant for thorough testing to find corner cases. However, over time, it grew to include such expensive tests.
This PR takes them out. It also allows us to bring back basictests to MinGW tests, since it is now cheap.

This is not an exercise in software engineering and making things prettier. This is a pragmatic change to improve our
test coverage and quality of life.

**I want basictests to build and run fast**.

### CODE REVIEW

**Comment by lemire** [comment]
The obvious objection to this PR is that one could pass parameters to basictests to select just the tests that one wants. But this would not solve the core issue: the tests that I am taking out are not what I consider "basic tests" (to quickly check whether the library is sane).

In fact, it would be easy to build quickly a small framework where one can pick and choose just some tests to run, using command line parameters. But there is elegance in having small executable files that do go on forever and that are logically coherent. When a test fails, you have less code to go through. It is faster to rebuild the binary and so forth.

**Comment by jkeiser** [comment]
Oh, I absolutely agree with this. basictests got out of hand, and it's largely my fault.

As far as selecting specific tests, I still think we need a test framework that lets us do that. But there's no reason not to have separate executables even then. And we're inching towards a framework with test_macros.h.

**Comment by jkeiser** [comment]
And I'm *super* stoked about removing the tests that take for gosh darn ever (the stream ones :)).

File: .github/workflows/mingw64-ci.yml [hunk 1]
```diff
@@ -47,10 +47,6 @@ jobs:
...
           cd build64
           cmake -DSIMDJSON_BUILD_STATIC=ON -DSIMDJSON_COMPETITION=OFF -DSIMDJSON_GOOGLE_BENCHMARKS=OFF -DSIMDJSON_ENABLE_THREADS=OFF ..
-          cmake --build . --target basictests numberparsingcheck stringparsingcheck errortests integer_tests pointercheck --verbose
-          ctest . -R stringparsingcheck  --output-on-failure
-          ctest . -R numberparsingcheck  --output-on-failure
-          ctest . -R errortests  --output-on-failure
-          ctest . -R integer_tests  --output-on-failure
-          ctest . -R pointercheck  --output-on-failure
+          cmake --build . --target parse_many_test jsoncheck basictests numberparsingcheck stringparsingcheck errortests integer_tests pointercheck --verbose
+          ctest . -R "(parse_many_test|jsoncheck|basictests|stringparsingcheck|numberparsingcheck|errortests|integer_tests|pointercheck)"  --output-on-failure
```
Comment (jkeiser) [inline .github/workflows/mingw64-ci.yml:51]: Would a --parallel on the build and test help make it go faster?

Comment (lemire) [reply .github/workflows/mingw64-ci.yml:51]: The thing appears to freeze when the tests are too intense and I am preoccupied that --parallel may make things worse. In any case, it is quite fast... a few seconds now...
